lib/benchpark/new_scaling.py

```diff
@@ -0,0 +1,398 @@
+                result.add_scalar_variable("process_problem_size", problem_sizes.prod)
+                result.add_scalar_variable("total_problem_size", total_problem_size)
+
+                return result
+            ScalingType.weak_scale = weak_scale
+
+        if mode == ScalingMode.Throughput:
+            @requires_experiment_variables("num_procs", "problem_sizes")
+            def throughput_scale(self):
+                """
+                Throughput scales the problem by increasing the per process size along each dimension in a
+                round-robin manner, starting with the minimum process dimension.
+                There are no changes to the number of processes.
+                """
+
+                num_procs = self.expr_vars.num_procs
+                problem_sizes = self.expr_vars.problem_sizes
+
+                if problem_sizes.ndims != num_procs.ndims:
+                    raise BenchparkError(
+                        f"problem_sizes dimensions {problem_sizes.dims} do not match num_procs dimensions {num_procs.dims}"
+                    )
+
+                num_exprs = int(self.spec.variants["scaling-iterations"][0]) - 1
+                scaling_factor = int(self.spec.variants["scaling-factor"][0])
+
+                start_dim = num_procs.min_dim
+                ndims = num_procs.ndims
+         
+                total_problem_size = [num_procs.prod[-1] * problem_sizes.prod[-1]]
+
+                for itr in range(num_exprs):
+                    dim = (start_dim + itr) % ndims
+
+                    problem_sizes.scale_dim(dim, lambda v: v * scaling_factor)
+                    total_problem_size.append(num_procs.prod[-1] * problem_sizes.prod[-1])
+
+                result = VariableDict()
+                result.add_scalar_variable("nprocs", num_procs.prod)
+                result.add_scalar_variable("process_problem_size", problem_sizes.prod)
+                result.add_scalar_variable("total_problem_size", total_problem_size)
+
+                return result
+            ScalingType.throughput_scale = throughput_scale
+
+    def register_required_variables(self):
+        self.required_vars.extend(["nprocs", "process_problem_size", "total_problem_size"])
+
+    return type(
+        "PerProcessDomainScaling",
+        (ScalingType,),
+        {
+            "register_required_variables": register_required_variables,
+        },
+    )
+
+
+def UsesGlobalDomains(*modes):
+    ScalingType = Scaling(*modes)
+
+    for mode in modes:
+        if mode == ScalingMode.Strong:
+            @requires_experiment_variables("num_procs", "problem_sizes")
+            def strong_scale(self):
+                """
+                Strong scales the problem by increasing the number of processes along each dimension,
+                starting with the minimum process dimension, in a round-robin manner.
+                There are no changes to the per process sizes.
```

Comment:
I was assuming if you increase the number of processes, that these types of experiments (i.e. ones that apply `UsesGlobalDomains`) were capable of splitting up the work automatically between the processes automatically. In that sense, I think

> There are no changes to the per process sizes.

isn't quite accurate: if the benchmark were not able to change the per-process problem size, I think strong scaling would not be possible (i.e. the total problem size would always increase when the # of processes was increased).